### ISSUE TITLE
JFrog free account will be deactivated on 2023-07-02

### ISSUE THREAD

**Issue by Ortham** [loot/loot#1872]
I've just received an email from JFrog notifying me that they're sunsetting their free accounts and that mine is due for deactivation on 2023-07-02, and all my data will be permanently removed. I only use the account to store LOOT org snapshot builds, so they will be affected.

JFrog's paid accounts start at $98/month to host it on AWS, GCP or Azure, or $150/month to use their cloud, so switching to a paid account isn't feasible.

Looking at the stats for the account, what's interesting is that more data is uploaded than is downloaded. This makes sense because nobody downloads most builds, but it also suggests that we're not really gaining all that much from having the builds available on Artifactory, so I think it's worth seeing what happens if we don't replace Artifactory with anything. If we want to share builds we can link to the GitHub Actions runs for people with GitHub accounts or upload the artifacts ad-hoc to e.g. Google Drive if we need to share them with people without GitHub accounts.

It's not as good as having an automated built-in "official" solution, and I'd like to explore signing artifacts as a way to verify authenticity so that there's at least a way to check if a file uploaded to some online drive is legitimate: we could give GitHub Actions a private key and publish the public key somewhere and the pipeline could create signature files to go with each artifact that gets created.

I don't intend to publish signatures for releases, since they get published on GitHub and I think it's good enough that anyone can cross-check file hashes of files downloaded elsewhere with the files on GitHub (or just use the files from GitHub) if they're not sure.

Here are some graphs from my JFrog account that give an idea of how much storage and bandwidth we use:

![image](https://github.com/loot/loot/assets/2430738/106af462-4bc7-47d0-82f1-6a14c8d1c5a5)
![image](https://github.com/loot/loot/assets/2430738/835681f8-6684-4289-9e01-17ab988784fa)
![image](https://github.com/loot/loot/assets/2430738/468b1754-cec8-44b1-9bac-4a4a18cc4ce7)
![image](https://github.com/loot/loot/assets/2430738/57f0ef3c-8ef9-43d7-bf1b-87e2c9d7c4a0)
![image](https://github.com/loot/loot/assets/2430738/3214cd00-60c4-46ed-933d-7efcb6ae0de8)

**Comment by Ortham** [comment]
I've implemented snapshot signing for the libloot and loot repositories, and removed publishing to Artifactory for them and the metadata-validator and python-libloot repos. The latter two don't have snapshot signing because I don't think it's worth it: I don't think I've ever had to share those artifacts with people without GitHub accounts.

I'll keep this issue open until I've confirmed that the Artifactory instance we've been using is gone.
